### DISCUSSION TITLE
Collect requirements for v2.5.0

### DISCUSSION THREAD

**Discussion by yanji09** [goharbor/harbor#15940] [Ideas]
Harbor team is planning for v2.5.0

If any features you would like to see in the next release, please input your ideas/requirements in this discussion.
If there's already an issue opened for a feature, please refer it here as well.

Thanks.

**Comment by Vad1mo** [comment]
I would like 2.5 to focus more on stability and resilience:
A perhaps **somewhat wild** claim from my side is that Harbor can't deal with storage sizes beyond > 1-2 TB. Especially on object storage.

1. GC Speedup and resilience.
https://github.com/goharbor/harbor/issues/15469
https://github.com/goharbor/harbor/issues/15822
https://github.com/goharbor/harbor/issues/15807
https://github.com/goharbor/harbor/issues/12948
https://github.com/goharbor/harbor/issues/15970



1. Replication resilience
I will lookup and add the open issues in that regard as a reference here.

**Reply by dkulchinsky** [reply]
Thank you for bringing this up @Vad1mo! I'd like to chime in and share our mutual concern, as our usage of Harbor grows we encounter perfromance and stability constraints that are concerning, in both of our environments (test & prod) GC has become inoperable due to the issues mentioned above.

I would love to see some focus to address these issues.

**Reply by juliohm1978** [reply]
@Vad1mo 

1-2 TB seems like a small threshold to start having problems. That's actually pretty concerning, IMO. Can you ellaborate on any problems you already had?

**Reply by dkulchinsky** [reply]
@juliohm1978, indeed it's a concern - if you take a look at the issues @Vad1mo linked I think you'll see what kind of problems we're hitting, in our experience it's mostly GC related, but we're also starting to see issues around retention policies execution - taking a long time and consuming considerable resources, for reference:
- https://github.com/goharbor/harbor/issues/14708

**Reply by dkulchinsky** [reply]
@Vad1mo, please also add:
- https://github.com/goharbor/harbor/issues/15970

to the GCs issue list above

**Comment by dkulchinsky** [comment]
I would also like to raise the feature request:

- https://github.com/goharbor/harbor/issues/15652

as our environments grow, we feel we need better control on how scheduled image scan-all are ran, specifically we would like the ability to filter our projects and/or repositories from these scans, for example: we have >20K artifacts that are image layer caches managed by our CI build system and scanning them is basically a waste of time and resources.

In addition, scanning images in a large environment can take up to 24 hours and more and consumes considerable resources, especially available jobservice workers, which "blocks" scans that are initiated by image pushes and require a scan result during CI to validate the new image being pushed, since scan-all basically consumes all available worker slots these scan-on-push jobs can be in the queue for many minutes, even hours.

so perhaps a more granular management of the workers utilization? for example that the scan-all job can't take more than X% of the overall available workers?

These are issues we're stumbling as our environment grow and I think worth addressing.

**Comment by Vad1mo** [comment]
Here is the summary of this discussion as an issue.

https://github.com/goharbor/harbor/issues/16010

**Comment by sixeela** [comment]
Implement a modulable permission system (custom groups/custom permissions per user)
